### ISSUE TITLE
PSR12.Classes.AnonClassDeclaration.CloseBraceSameLine false positive for anon class passed as function argument

### ISSUE THREAD

**Issue by martinssipenko** [squizlabs/PHP_CodeSniffer#2621]
When using anonymous class the following should be allowed, currently it is reported as error.

```
Closing class brace must be on a line by itself
(PSR12.Classes.AnonClassDeclaration.CloseBraceSameLine)
```

```php
$foo->bar(
    new class implements Bar {
        // ...
    },
);
```
